### PR TITLE
Add deployment list and follower

### PR DESCRIPTION
**Pull request by johnsudaar**

### CODE REVIEW

File: test/Deployments/listener.test.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        })
+      })
+
+      expect(stub.called).to.be.true
+      expect(stub.getCall(0).args[0]).to.deep.eq({
+        deployment: "123e4567-e89b-12d3-a456-426655440000"
+      })
+    })
+
+    it("shoult call the correct callback when we get a 'log' message", () => {
```
Comment (ichbinkour) [inline test/Deployments/listener.test.js]:
```suggestion
    it("should call the correct callback when we get a 'log' message", () => {
```

File: test/Deployments/listener.test.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      })
+
+      expect(stub.called).to.be.true
+      expect(stub.getCall(0).args[0]).to.deep.eq({
+        id: "test-id",
+        content: "Hey !",
+      })
+    })
+
+    it("shoult call the correct callback when we get a 'status' message", () => {
```
Comment (ichbinkour) [inline test/Deployments/listener.test.js]:
```suggestion
    it("should call the correct callback when we get a 'status' message", () => {
```

File: src/Deployments/listener.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  close() {
+    if(this._ws) {
+      this._ws.close()
+      this._ws = null
+    }
+  }
+
+  /**
+   * Setup a callback that will be called when the connection is closed.
+   * @param {function()} callback Callback to call when the connction is closed
```
Comment (ichbinkour) [inline src/Deployments/listener.js]:
```suggestion
   * @param {function()} callback Callback to call when the connection is closed
```

File: src/Deployments/index.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+ * @desc
+ * Can take the following values:
+ * - **building**: when the buildpack is chosen and executed
+ * - **pushing**: the generated Docker image is pushed to our registry
+ * - **starting**: the order to start your app has been transmitted to our scheduler
+ * - **success**: the application has correctly started, end of deployment
+ * - **crashed**-error: the application failed to boot, have a look to the logs
+ * - **timeout-error**: the application has not started in 60 seconds
+ * - **build-error**: something went wrong during the buildpack execution, have a look at deployments logs
+ * - **aborted**: if the connection is broken between the pusher and our infrastructure, the deployment is considered as canceled.
```
Comment (ichbinkour) [inline src/Deployments/index.js]: Isn't it `**crashed-error**` ?

File: src/Deployments/index.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * @typedef {String} DeploymentStatus
+ * @desc
+ * Can take the following values:
+ * - **building**: when the buildpack is chosen and executed
+ * - **pushing**: the generated Docker image is pushed to our registry
+ * - **starting**: the order to start your app has been transmitted to our scheduler
+ * - **success**: the application has correctly started, end of deployment
+ * - **crashed**-error: the application failed to boot, have a look to the logs
```
Comment (johnsudaar) [inline src/Deployments/index.js]:
```suggestion
 * - **crashed-error**: the application failed to boot, have a look to the logs
```

**Comment by ichbinkour** [comment]
LGTM

**Comment by Soulou** [comment]
It's really neat, LGTM

**Pull request merged**